PocketKit/Sources/PocketKit/Item/ItemViewController.swift

```diff
@@ -68,7 +144,53 @@ class ItemViewController: UIViewController {
     }
 
     @objc
-    private func showOverflow() {
-        delegate?.itemViewControllerDidTapOverflowButton(self)
+    private func showReaderSettings() {
+        delegate?.itemViewControllerDidTapReaderSettings(self)
+    }
+}
+
+// MARK: - Item Actions
+
+extension ItemViewController {
+    private func favorite() {
+        guard let item = itemHost.item else {
+            return
+        }
+
+        source.favorite(item: item)
```

Comment:
Curious on your thoughts here: for something like `showReaderSettings`, we tell the _delegate_ to perform the action, but here, the view controller itself is unfavoriting the item. Do we want the current instance of `ItemViewController` to (un)favorite the item (as well as the other actions), or would we rather push that off to the delegate?

Reply:
I struggled with this issue too. I settled on calling the `source` method here in the view controller. The reason being that the VC's responsible for performing actions but it delegates the work of presenting or showing other views.